File: app/src/main/java/com/shui520it/access/funcion/action/ShuiBaseAction.java
package com.shui520it.access.funcion.action;

import android.os.Build;
import android.view.accessibility.AccessibilityNodeInfo;

import androidx.annotation.Nullable;
import androidx.annotation.RequiresApi;

import java.util.Collection;
import java.util.List;

/**
 * @author shuimu{lwp}
 * @time 2019/8/14  11:59
 * @desc
 */
public abstract class ShuiBaseAction<T> {
    /**
     * 是包含还必须相等;
     */
    public boolean isEquals = true;
    public T acitonKey;

    /**
     * 子类检查是否 就是 正寻找的内容对象
     */
    public boolean checkOk(AccessibilityNodeInfo thisInfo) {
        return true;
    }

    /**
     * 返回一个 寻找内容
     *
     * @param root 被寻找的父类
     * @return
     */
    @Nullable
    public AccessibilityNodeInfo findFirst(AccessibilityNodeInfo root) {
        return null;
    }

    /**
     * 返回一组寻找内容
     *
     * @param root 被寻找的父类
     * @return
     */

    @RequiresApi(api = Build.VERSION_CODES.JELLY_BEAN_MR2)
    @Nullable
    public List<AccessibilityNodeInfo> findAll(AccessibilityNodeInfo root) {
        return null;
    }

    /**
     * 集合是否是空的
     */
    public static boolean isEmpty(Collection list) {
        return list == null || list.size() == 0;
    }
}

File: app/src/main/java/com/shui520it/access/funcion/action/ShuiBaseAction.kt
package com.shui520it.access.funcion.action

import android.view.accessibility.AccessibilityNodeInfo

/**
 * @author shuimu{lwp}
 * @time  2021/2/1 13:44
 * @desc
 */
abstract class ShuiBaseAction<T> {
    /**
     * 包含 or 相等
     */
    open var isEquals = true

    /**
     * 处理的数据
     */
   open var actionKey: T? = null

    /**
     * 检查是否符合要求，默认true
     */
    open fun checkOk(info: AccessibilityNodeInfo) = true

    /**
     * 返回集合的第一个
     */
    open fun findFirst(info: AccessibilityNodeInfo): AccessibilityNodeInfo? = null

    /**
     * 返回集合
     */
    open fun findAll(info: AccessibilityNodeInfo): List<AccessibilityNodeInfo>? = null
}